esp_flasher/gui.py

```
@@ -8,83 +7,62 @@
-            if self._show_logs:
-                argv.append('--show-logs')
-            run_esp_flasher(argv)
+            
+            # Call with skip_logs=True so it returns after flashing
+            run_esp_flasher(argv, skip_logs=True)
+            
+            # Emit finished signal
+            if self.finished:
+                self.finished.emit()
+                
         except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)